modules/openfoam/src/OpenFOAM_Types.f90

```diff
@@ -2427,151 +2386,109 @@ SUBROUTINE OpFM_CopyInput( SrcInputData, DstInputData, CtrlCode, ErrStat, ErrMsg
       CALL SetErrStat(ErrID_Fatal, 'Error allocating DstInputData%forceNodesChord.', ErrStat, ErrMsg,RoutineName)
       RETURN
     END IF
-    DstInputData%c_obj%forceNodesChord_Len = SIZE(DstInputData%forceNodesChord)
-    IF (DstInputData%c_obj%forceNodesChord_Len > 0) &
-          DstInputData%c_obj%forceNodesChord = C_LOC( DstInputData%forceNodesChord( i1_l ) )
+    DstInputData%C_obj%forceNodesChord_Len = SIZE(DstInputData%forceNodesChord)
+    IF (DstInputData%C_obj%forceNodesChord_Len > 0) &
+          DstInputData%C_obj%forceNodesChord = C_LOC( DstInputData%forceNodesChord( i1_l ) )
   END IF
     DstInputData%forceNodesChord = SrcInputData%forceNodesChord
 ENDIF
  END SUBROUTINE OpFM_CopyInput
 
- SUBROUTINE OpFM_DestroyInput( InputData, ErrStat, ErrMsg, DEALLOCATEpointers )
+ SUBROUTINE OpFM_DestroyInput( InputData, ErrStat, ErrMsg )
   TYPE(OpFM_InputType), INTENT(INOUT) :: InputData
   INTEGER(IntKi),  INTENT(  OUT) :: ErrStat
   CHARACTER(*),    INTENT(  OUT) :: ErrMsg
-  LOGICAL,OPTIONAL,INTENT(IN   ) :: DEALLOCATEpointers
   
   INTEGER(IntKi)                 :: i, i1, i2, i3, i4, i5 
-  LOGICAL                        :: DEALLOCATEpointers_local
   INTEGER(IntKi)                 :: ErrStat2
   CHARACTER(ErrMsgLen)           :: ErrMsg2
   CHARACTER(*),    PARAMETER :: RoutineName = 'OpFM_DestroyInput'
 
   ErrStat = ErrID_None
   ErrMsg  = ""
 
-  IF (PRESENT(DEALLOCATEpointers)) THEN
-     DEALLOCATEpointers_local = DEALLOCATEpointers
-  ELSE
-     DEALLOCATEpointers_local = .true.
-  END IF
-  
 IF (ASSOCIATED(InputData%pxVel)) THEN
- IF (DEALLOCATEpointers_local) &
   DEALLOCATE(InputData%pxVel)
-  InputData%pxVel => NULL()
```

Comment:
Given the comment from Dr. Fortran (comment from @bjonkman above), we may need this as well here.